src/pages/search.tsx

```diff
@@ -84,26 +84,36 @@ const Search: NextPage<SearchProps> = ({ translations }): JSX.Element => {
   // After hydration, Next.js will trigger an update to provide the route parameters
   // in the query object. @see https://nextjs.org/docs/routing/dynamic-routes#caveats
   useEffect(() => {
-    if (router.isReady) {
-      if (router.query.q || router.query.query) {
-        let query = router.query.q as string;
-        if (router.query.query) {
-          query = router.query.query as string;
-        }
-        setSearchQuery(query);
-      }
-      if (router.query.page) {
-        setCurrentPage(Number(router.query.page));
-      }
-      if (router.query.languages) {
-        setSelectedLanguages(router.query.languages as string);
-      }
-      if (router.query.translations) {
-        setSelectedTranslations(router.query.translations as string);
-      }
+    if (router.isReady && !isContentModalOpen) {
       focusInput();
     }
-  }, [focusInput, router]);
+  }, [focusInput, router, isContentModalOpen]);
+
+  useEffect(() => {
+    if (router.query.q || router.query.query) {
+      let query = router.query.q as string;
+      if (router.query.query) {
+        query = router.query.query as string;
+      }
+      setSearchQuery(query);
+    }
+
+    if (router.query.page) {
+      setCurrentPage(Number(router.query.page));
+    }
+    if (router.query.languages) {
+      setSelectedLanguages(router.query.languages as string);
+    }
+    if (router.query.translations) {
+      setSelectedTranslations(router.query.translations as string);
+    }
+  }, [
+    router?.query?.q,
+    router?.query?.query,
+    router?.query?.page,
+    router?.query?.languages,
+    router?.query?.translations,
```

Comment:
Why do we need `?.` after `router`?

Reply:
Maybe I am overprotective here. 
I will make the cleanup change.